src/lambda/expression.rs

```diff
@@ -168,6 +172,20 @@ impl Expression {
             }
           }
         }
+        Self::Merge(subs) => {
```

Comment:
could you provide an example of problem you're solving with this? I'm not sure this is part of the original issue